test/valid-data/type-intersection-union-enum/schema.json

```diff
@@ -0,0 +1,31 @@
+{
+  "$ref": "#/definitions/MyObject",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "definitions": {
+    "MyObject": {
+      "additionalProperties": false,
+      "properties": {
+        "field1": {
+          "enum": [
+            "foo",
+            "bar"
+          ],
+          "type": "string"
+        },
+        "field2": {
```

Comment:
Can you make it so that field2 extends field1? I think that should be possible with `oneOf`, no?

Reply:
updated to use `oneOf`, have a look 😄

Reply:
There should be a ref to field1 in there now.

Reply:
hmm, okay, is the names somewhere in `type`? plus do I need to take care of possible dereferencing here, or is it handled centrally?

Reply:
If you create an alias type, everything else will be handled for you. Take a look at some of the other parsers.